src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -351,14 +351,26 @@ private IEventReferenceExpression CreateBoundEventAccessOperation(BoundEventAcce
 
         private IEventAssignmentExpression CreateBoundEventAssignmentOperatorOperation(BoundEventAssignmentOperator boundEventAssignmentOperator)
         {
-            IEventSymbol @event = boundEventAssignmentOperator.Event;
-            Lazy<IOperation> eventInstance = new Lazy<IOperation>(() => Create(boundEventAssignmentOperator.Event.IsStatic ? null : boundEventAssignmentOperator.ReceiverOpt));
+            SyntaxNode syntax = boundEventAssignmentOperator.Syntax;
+            Lazy<IEventReferenceExpression> eventReference = new Lazy<IEventReferenceExpression>(() =>
+            {
+                IEventSymbol @event = boundEventAssignmentOperator.Event;
+                Lazy<IOperation> instance = new Lazy<IOperation>(() => Create(boundEventAssignmentOperator.Event.IsStatic ? null : boundEventAssignmentOperator.ReceiverOpt));
```

Comment:
Actually, it seems the conditional is necessary, since the receiver will not be null for static event reference:
http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Binder/Binder_Expressions.cs,5809